code/obj/artifacts/artifact_items/dimensional_key.dm

```diff
@@ -0,0 +1,394 @@
+				inner_door = locate() in src.fissure_region.turf_at(13, 19)
+				inner_door.outer_entrance_spawned = TRUE
+				adj_entr_turfs = block(entrance.x - 1, entrance.y - 1, entrance.z, entrance.x - 1, entrance.y + 1, entrance.z)
+
+		entrance.density = FALSE
+		var/obj/art_fissure_objs/door/outer_door = new(entrance)
+		outer_door.set_dir(get_dir(outer_door, user))
+		outer_door.linked_door = inner_door
+		inner_door.set_dir(get_dir(outer_door, user))
+		inner_door.linked_door = outer_door
+		fissure_entr.reachable_turfs += adj_entr_turfs
+		for (var/turf/T as anything in fissure_entr.reachable_turfs)
+			if (!T.reachable_turfs)
+				T.reachable_turfs = list()
+			T.reachable_turfs += fissure_entr
+		var/area/artifact_fissure/A = get_area(inner_door)
+		A.update_visual_mirrors(entrance, entrance_dir)
+		new /obj/art_fissure_objs/mirror_update_dummy(fissure_entr, entrance, entrance_dir)
+		entrance.icon = 'icons/turf/floors.dmi'
+		entrance.icon_state = "darkvoid"
+		entrance.opacity = TRUE
+		entrance.name = "Thick void mist"
+		entrance.desc = "Void mist thick enough that you can't see through it.. How did this get here?"
+		RL_UPDATE_LIGHT(entrance)
+
+		logTheThing(LOG_STATION, src, "Dimensional key artifact door created at [log_loc(outer_door)] by [key_name(user)].")
+
+/datum/artifact/dimensional_key
+	associated_object = /obj/item/artifact/dimensional_key
+	type_name = "Dimensional key"
+	type_size = ARTIFACT_SIZE_TINY
+	rarity_weight = 200
+	validtypes = list("eldritch", "precursor")
+	react_xray = list(73, 90, 38, 4, "ANOMALOUS")
+	examine_hint = "It kinda looks like it's supposed to be inserted into something."
+
+	effect_activate(obj/O)
+		. = ..()
+		if (.)
+			return TRUE
+		var/obj/item/artifact/dimensional_key/artkey = O
+		var/datum/mapPrefab/allocated/allocated = get_singleton(/datum/mapPrefab/allocated/artifact_fissure)
+		artkey.fissure_region = allocated.load()
+		var/turf/T = artkey.fissure_region.get_center()
+		var/area/artifact_fissure/fissure_area = get_area(T)
+		fissure_area.fissure_region = artkey.fissure_region
+
+		mirrored_physical_zone_created = TRUE
+
+/****** Supporting items/atoms/etc. *******/
+
+ABSTRACT_TYPE(/obj/art_fissure_objs)
+/obj/art_fissure_objs
+	name = ""
+	desc = ""
+	anchored = ANCHORED_ALWAYS
+
+ABSTRACT_TYPE(/obj/art_fissure_objs/cross_dummy)
+/obj/art_fissure_objs/cross_dummy
+	invisibility = INVIS_ALWAYS
+	var/turf/exit_turf
+	var/required_dir
+
+	New(newLoc, turf/exit)
+		..()
+		src.exit_turf = exit
+
+	disposing()
+		src.exit_turf = null
+		..()
+
+	Crossed(atom/movable/AM)
+		if (istype(AM, /obj/projectile))
+			var/obj/projectile/P = AM
+			var/obj/projectile/new_proj = initialize_projectile(src.exit_turf, P.proj_data, P.xo, P.yo, P.shooter)
+			new_proj.travelled = P.travelled
+			new_proj.launch()
+			P.die()
+		else if (AM.dir == src.required_dir && !istype(AM, /obj/art_fissure_objs/door) && !istype(AM, /obj/art_fissure_objs/cross_dummy))
+			AM.set_loc(src.exit_turf)
```

Comment:
Ideally this should either use `warptarget` or duplicate the weird double `set_loc` behaviour warp targets do to fix the glide animation.

Reply:
Nice, with the double set_loc, seems it does let you see past the walls at the entry/exit for a very small moment, if that's an issue

Reply:
Hmm, I thiink that's better than the jerky teleport? Up to you though, I can't think of a way to get the best of both as it were.

Reply:
kk, yeah I think it does look better